app/main.py

```diff
@@ -1 +1,37 @@
-# write your code here
+from __future__ import annotations
+from typing import Union
+
+
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.hidden = False
+        self.health = health
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{"
+                f"Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(other: Union[Carnivore, Herbivore]) -> None:
+        if isinstance(other, Carnivore):
+            return
+        if isinstance(other, Herbivore) and other.hidden:
+            return
+        other.health -= 50
+        if other.health <= 0:
+            Animal.alive.pop(Animal.alive.index(other))
```

Comment:
Also at 37 line ` Animal.alive.pop(Animal.alive.index(other))`

Simplify this line. We access class attributes through class name and it's better to use **remove when value is given** and **del when an index is given**. You do unnecessary work here